ci/prep_cython_cache.sh

```diff
@@ -1,43 +1,73 @@
 #!/bin/bash
 
 ls "$HOME/.cache/"
+
+PYX_CACHE_DIR="$HOME/.cache/pyxfiles"
+pyx_file_list=`find ${TRAVIS_BUILD_DIR} -name "*.pyx"`
+pyx_cache_file_list=`find ${PYX_CACHE_DIR} -name "*.pyx"`
+
 CACHE_File="$HOME/.cache/cython_files.tar"
 
+# Clear the cython cache 0 = NO, 1 = YES
 clear_cache=0
+
+pyx_files=`echo "$pyx_file_list" | wc -l`
+pyx_cache_files=`echo "$pyx_cache_file_list" | wc -l`
```

Comment:
this will not work. you would need to compare .pyx file sizes I think to make this reliable. Just adding / removing a file is not good enough at all. Most often we have a change in actual contents.